src/VisualStudio/CSharp/Impl/ProjectSystemShim/CSharpProjectShim.ICSharpProjectSite.cs

```diff
@@ -121,13 +120,12 @@ public void OnModuleRemoved(string filename)
 
         public int GetValidStartupClasses(IntPtr[] classNames, ref int count)
         {
+            var project = Workspace.CurrentSolution.GetProject(VisualStudioProject.Id)!;
```

Comment:
We have GetRequiredProject too that will throw a helpful message if it's not there.